### DISCUSSION TITLE
Auto Saving All Incoming/Outgoing Chat messages

### DISCUSSION THREAD

**Discussion by RevealedSoulEven** [rodit/SnapMod#114] [Ideas & Feature Suggestions]
As you know it's quite irritating to tap on each and every message while chatting with a friend, and it's a bit time consuming to do too. So it would be better if there's an option to automatically save all incoming and outgoing chat messages/media files(not snaps)/voice messages/stickers and all in the chat so that we won't need to tap on it to save.

Yes btw you can add another switch to save all incoming snaps in the chat too but that's not the real concern here. It's regarding when we chat on snapchat we need to tap on every single message, so getting rid of it.

In addition, I want to add that this feature is alredy available in [SnapTrap](https://github.com/TheGreyRaven/SnapTrap) .

Hope to see this feature soon. Thank You! And really appreciate your work❤️

**Comment by rodit** [comment]
There's no code in the SnapTrap repo that suggests it has this feature - it should be quite easy, however, to implement.

**Reply by NMBlanco** [reply]
> There's no code in the SnapTrap repo that suggests it has this feature - it should be quite easy, however, to implement.

I think he meant SnapTools. The automatic message saving was a feature there

**Reply by RevealedSoulEven** [reply]
I haven't tried anything yet. But in SnapTrap website it is written. But yea @rodit please implement that🙏 It would be very useful

**Comment by Magna-Graecia** [comment]
This feature is already implemented, it's just not working at the moment. "Don't Release Messages" is basically supposed to do what you're requesting here. Text, photos/videos, and audio notes are all considered messages, so if the "Don't Release Messages" setting was working, then I'm sure they would all be auto-saved when this setting was enabled. It's just a matter of figuring out how to make it work.

I never used SnapTools, but SnapFreedom had this feature. It would literally save all messages, including photos, videos, audio notes, etc with the grey box as if you were tapping on them yourself, but it was automatic. The other person could see it though, so it's not like they were being stealth saved. You can basically accomplish the same thing by just changing the chat setting to delete after 24 hours, but then again they only stay for 24 hours instead of forever. Personally 24 hours is enough for me. Having too many chats filled with permanently saved messages will slow down your Snapchat.

**Comment by RevealedSoulEven** [comment]
Yeaaa. But actually that feature is eagerly required, when will it get released? Any idea?

**Comment by RevealedSoulEven** [comment]
Btw are you focusing on it @rodit ?

**Comment by NMBlanco** [comment]
Bro don't force him 😭

**Comment by rodit** [comment]
Available in 1.6.2 which will appear in a few mins.

Needs some testing so use this discussion to report any bugs (or open an issue) after testing it. It **requires** Snapchat 11.78.1.39 (just released yesterday, not even in my play store yet).

**Reply by Magna-Graecia** [reply]
Working beautifully on my end! Great job, rodit!

**Reply by RevealedSoulEven** [reply]
Waaowwww it's such a great feature. I am also a modder but you guys are fantabulous awesome. 
Great Job developers
